drivers/ethernet/eth_sam_gmac.c

```diff
@@ -1860,17 +1792,31 @@ static void eth0_iface_init(struct net_if *iface)
 #ifdef CONFIG_SOC_SAMA7G54
 		| GMAC_NCFGR_DBW(1)  /* Data Bus Width. Must always be written to ‘1’ */
 #endif
-		| GMAC_NCFGR_RXCOEN  /* Receive Checksum Offload Enable */
-		| GMAC_MAX_FRAME_SIZE;
-	result = gmac_init(cfg->regs, gmac_ncfgr_val);
+		| GMAC_NCFGR_RXCOEN; /* Receive Checksum Offload Enable */
+	switch (cfg->max_frame_size) {
+	case 1518:
+		break;
+	case 1536:
+		gmac_ncfgr_val |= GMAC_NCFGR_MAXFS;
+		break;
```

Comment:
the difference between these two is that 1536 would allow for packets with a vlan tag, right? If so, it should be chosen based on `CONFIG_NET_VLAN` and not the dt prop, so that prop could be removed.

Reply:
you can also remove it from the dt binding yaml

Reply:
remove it directly.